arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -2100,23 +2103,19 @@ void RestReplicationHandler::handleCommandAddFollower() {
       referenceChecksum = std::to_string(num);
     }
 
-    auto result = col->compareChecksums(checksum, referenceChecksum);
-
-    if (result.fail()) {
-      auto errorNumber = result.errorNumber();
-      rest::ResponseCode code;
-      if (errorNumber == TRI_ERROR_REPLICATION_WRONG_CHECKSUM ||
-          errorNumber == TRI_ERROR_REPLICATION_WRONG_CHECKSUM_FORMAT) {
-        code = rest::ResponseCode::BAD;
-      } else {
-        code = rest::ResponseCode::SERVER_ERROR;
-      }
-      generateError(code, errorNumber, result.errorMessage());
+    if (!checksumSlice.isEqualString(referenceChecksum)) {
+      const std::string checksum = checksumSlice.copyString();
+      LOG_TOPIC(WARN, Logger::REPLICATION) << "Cannot add follower, mismatching checksums. "
+       << "Expected: " << referenceChecksum << " Actual: " << checksum;
+      generateError(rest::ResponseCode::BAD, TRI_ERROR_REPLICATION_WRONG_CHECKSUM,
+                    "'checksum' is wrong. Expected: "
+                    + referenceChecksum
+                    + ". Actual: " + checksum);
       return;
     }
   }
 
-  col->followers()->add(followerId.copyString());
+  col->followers()->add(followerIdSlice.copyString());
```

Comment:
shouldn't this be `col->followers()->add(followerId);`

Reply:
Yes, it should.